pkg/controller/master/storagenetwork/storage_network.go

```diff
@@ -0,0 +1,844 @@
+			return true
+		}
+
+		settingCopy := setting.DeepCopy()
+		if err := h.removeOldNad(settingCopy); err != nil {
+			logrus.Warnf("remove old nad error %v", err)
+			h.settingsController.EnqueueAfter(setting.Name, 5*time.Second)
+			return true
+		}
+
+		// finish
+		if _, err := h.setConfiguredCondition(settingCopy, true, ReasonCompleted, ""); err != nil {
+			logrus.Warnf("update status error %v", err)
+		}
+		return true
+	}
+	return false
+}
+
+// true: all detach
+func (h *Handler) checkLonghornVolumeDetach() (bool, error) {
+	volumes, err := h.longhornVolumeCache.List(util.LonghornSystemNamespaceName, labels.Everything())
+	if err != nil {
+		logrus.Warnf("volume error %v", err)
+		return false, err
+	}
+
+	for _, volume := range volumes {
+		if volume.Status.State != "detached" {
+			logrus.Infof("volume state: %v", volume.Status.State)
+			return false, nil
+		}
+	}
+	return true, nil
+}
+
+func (h *Handler) checkPrometheusStatusAndStart() bool {
+	// check prometheus cattle-monitoring-system/rancher-monitoring-prometheus replica
+	prometheus, err := h.prometheusCache.Get(CattleMonitoringSystemNamespace, RancherMonitoringPrometheus)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			logrus.Infof("prometheus not found. skip")
+			return true
+		}
+		logrus.Warnf("prometheus get error %v", err)
+		return false
+	}
+
+	// check started or not
+	if replicasStr, ok := prometheus.Annotations[ReplicaStorageNetworkAnnotation]; ok {
+		logrus.Infof("current prometheus replicas: %v", *prometheus.Spec.Replicas)
+		logrus.Infof("start prometheus")
+		prometheusCopy := prometheus.DeepCopy()
+		// ignore error, and keep original replica
+		if replicas, err := strconv.Atoi(replicasStr); err == nil {
+			*prometheusCopy.Spec.Replicas = int32(replicas)
+		}
+		delete(prometheusCopy.Annotations, ReplicaStorageNetworkAnnotation)
+
+		if _, err := h.prometheus.Update(prometheusCopy); err != nil {
+			logrus.Warnf("prometheus update error %v", err)
+			return false
+		}
+		return true
+	}
+
+	return true
+}
+
+func (h *Handler) checkAltermanagerStatusAndStart() bool {
+	// check alertmanager cattle-monitoring-system/rancher-monitoring-alertmanager replica
+	alertmanager, err := h.alertmanagerCache.Get(CattleMonitoringSystemNamespace, RancherMonitoringAlertmanager)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			logrus.Infof("Alertmanager not found. skip")
+			return true
+		}
+		logrus.Warnf("alertmanager get error %v", err)
+		return false
+	}
+
+	// check started or not
+	if replicasStr, ok := alertmanager.Annotations[ReplicaStorageNetworkAnnotation]; ok {
+		logrus.Infof("current alertmanager replicas: %v", *alertmanager.Spec.Replicas)
+		logrus.Infof("start alertmanager")
+		alertmanagerCopy := alertmanager.DeepCopy()
+		// ignore error, and keep original replica
+		if replicas, err := strconv.Atoi(replicasStr); err == nil {
+			*alertmanagerCopy.Spec.Replicas = int32(replicas)
+		}
```

Comment:
If someone modifies the annotation, it will cause this error here.
and to make the process could keep going, we should leave it as the original config and delete the annatation.